src/plugins/console/public/application/containers/editor/monaco/hooks/use_register_keyboard_commands.ts

```diff
@@ -0,0 +1,160 @@
+import { IDisposable } from 'monaco-editor';
+
+interface RegisterKeyboardCommandsParams {
+  /** The current Monaco editor instance. */
+  editor: monaco.editor.IStandaloneCodeEditor;
+  /** Function for sending the selected request(s). */
+  sendRequest: () => void;
+  /** Function for indenting the selected request(s). */
+  autoIndent: () => void;
+  /** Function that returns the documentation link for the selected request. */
+  getDocumentationLink: () => Promise<string | null> | undefined;
+  /** Function for moving the cursor to the previous request edge. */
+  moveToPreviousRequestEdge: () => void;
+  /** Function for moving the cursor to the next request edge. */
+  moveToNextRequestEdge: () => void;
+}
+
+const SEND_REQUEST_ACTION_ID = 'sendRequest';
+const AUTO_INDENT_ACTION_ID = 'autoIndent';
+const OPEN_DOCS_ACTION_ID = 'openDocs';
+const MOVE_UP_ACTION_ID = 'moveUp';
+const MOVE_DOWN_ACTION_ID = 'moveDown';
+const MOVE_TO_LINE_ACTION_ID = 'moveToLine';
+
+/**
+ * Hook that returns a function for registering keyboard commands in the editor.
+ *
+ * @param params The {@link RegisterKeyboardCommandsParams} to use.
+ */
+export const useKeyboardCommandsUtils = () => {
+  const sendRequestAction = useRef<IDisposable | null>(null);
+  const autoIndentAction = useRef<IDisposable | null>(null);
+  const openDocsAction = useRef<IDisposable | null>(null);
+  const moveToPreviousAction = useRef<IDisposable | null>(null);
+  const moveToNextAction = useRef<IDisposable | null>(null);
+  const moveToLineAction = useRef<IDisposable | null>(null);
+
+  const disposeAllActions = () => {
+    if (sendRequestAction.current) {
+      sendRequestAction.current.dispose();
+    }
+    if (autoIndentAction.current) {
+      autoIndentAction.current.dispose();
+    }
+    if (openDocsAction.current) {
+      openDocsAction.current.dispose();
+    }
+    if (moveToPreviousAction.current) {
+      moveToPreviousAction.current.dispose();
+    }
+    if (moveToNextAction.current) {
+      moveToNextAction.current.dispose();
+    }
+    if (moveToLineAction.current) {
+      moveToLineAction.current.dispose();
+    }
+  };
+
+  const registerKeyboardCommands = (params: RegisterKeyboardCommandsParams) => {
+    const {
+      editor,
+      sendRequest,
+      autoIndent,
+      getDocumentationLink,
+      moveToPreviousRequestEdge,
+      moveToNextRequestEdge,
+    } = params;
+
+    const openDocs = async () => {
+      const documentation = await getDocumentationLink();
+      if (!documentation) {
+        return;
+      }
+      window.open(documentation, '_blank');
+    };
+
+    disposeAllActions();
+
+    sendRequestAction.current = editor.addAction({
+      id: SEND_REQUEST_ACTION_ID,
+      label: i18n.translate('console.keyboardCommandActionLabel.sendRequest', {
+        defaultMessage: 'Send request',
+      }),
+      // eslint-disable-next-line no-bitwise
+      keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.Enter],
```

Comment:
could you please explain why we need to use a bitwise operator here?

Reply:
It seems that in Monaco, to create a keybinding that is a combination of multiple keys, you need to use a bitwise operator - see https://github.com/microsoft/monaco-editor/issues/42.